samples/nrf9160/serial_lte_modem/src/slm_at_icmp.c

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+#include <logging/log.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <string.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+#include <modem_info.h>
+#include "slm_at_icmp.h"
+
+LOG_MODULE_REGISTER(icmp, CONFIG_SLM_LOG_LEVEL);
+
+#define INVALID_SOCKET		-1
+#define ICMP_MAX_URL		128
+
+#define THREAD_STACK_SIZE	KB(1)
+#define THREAD_PRIORITY		K_LOWEST_APPLICATION_THREAD_PRIO
+
+#define ICMP			0x01
+#define ICMP_ECHO_REQ		0x08
+#define ICMP_ECHO_REP		0x00
+#define IP_PROTOCOL_POS		0x09
+
+/*
+ * Known limitation in this version
+ * - at RRC_Idle, the first Ping gets timeout
+ * - IPv6 support
+ */
+
+/**@brief List of supported AT commands. */
+enum slm_icmp_at_cmd_type {
+	AT_ICMP_PING,
+	AT_ICMP_MAX
+};
+
+/**@ ICMP Ping command arguments */
+static struct ping_argv
+{
+	struct addrinfo *src;   /* Localhost public address */
+	struct addrinfo *dest;  /* Destination for the ping */
+	int len;
+	int waitms;
+	int count;
+	int interval;
+} argv;
```

Comment:
I wonder, lone `argv` is well known as a function parameter of specific type (char**), I guess it could be confusing when encountered in code with a different type. We could use some specialized name here (`ping_argv` sounds reasonably, just as the struct type).

Reply:
Updated to

> static struct ping_argv_t {
> 	struct addrinfo *src;   /* Localhost public address */
> 	struct addrinfo *dest;  /* Destination for the ping */
> 	int len;
> 	int waitms;
> 	int count;
> 	int interval;
> } ping_argv;